src/cares_wrap.cc

```diff
@@ -714,6 +719,9 @@ class QueryWrap : public AsyncWrap {
       extra
     };
     const int argc = arraysize(argv) - extra.IsEmpty();
+    static const std::string trace_name = name_;
+    TRACE_EVENT_NESTABLE_ASYNC_END0("node,node.dns", trace_name.c_str(), this);
```

Comment:
This looks odd … this means that `trace_name` is initialized once and not changed afterwards? What if this function gets called with instances of QueryWrap that have different names?

I think things might end up being easier if you store the original `const char*` itself on the class and always pass that to the trace macros. That might require the string to be given at compile time, but I’d think that’s okay.

Reply:
Thanks for pointing it out. I will need to fix it.